src/main.cpp

```diff
@@ -2916,10 +2935,54 @@ int main(int arg_count, char const **arg_ptr) {
 		show_timings(checker, &global_timings);
 	}
 
-	if (run_output) {
-		String exe_name = path_to_string(heap_allocator(), build_context.build_paths[BuildPath_Output]);
-		defer (gb_free(heap_allocator(), exe_name.text));
+	String exe_name = path_to_string(heap_allocator(), build_context.build_paths[BuildPath_Output]);
+	defer (gb_free(heap_allocator(), exe_name.text));
+
+	if (build_context.export_dependencies_file.len > 0) {
+		gbFile f = {};
+		char * fileName = (char *)build_context.export_dependencies_file.text;
+		gbFileError err = gb_file_open_mode(&f, gbFileMode_Write, fileName);
+		if (err != gbFileError_None) {
+			gb_printf_err("Failed to export dependencies to: %s\n", fileName);
+			exit_with_errors();
+			return 1;
+		}
+		defer (gb_file_close(&f));
+
+		gb_fprintf(&f, "%.*s:", LIT(exe_name));
+
+		isize current_line_length = exe_name.len + 1;
+
+		for (isize i = parser->packages.count-1; i >= 0; i--) {
+			AstPackage *pkg = parser->packages[i];
+			for (isize j = pkg->files.count-1; j >= 0; j--) {
+				AstFile *file = pkg->files[j];
 
+				/* Arbitrary line break value. Maybe make this better? */
+				if (current_line_length >= 80-2) {
+					gb_file_write(&f, " \\\n ", 4);
+					current_line_length = 1;
+				}
+
+				gb_file_write(&f, " ", 1);
+				current_line_length++;
+
+				for (isize k = 0; k < file->fullpath.len; k++) {
+					char part = file->fullpath.text[k];
+					if (part == ' ') {
+						gb_file_write(&f, "\\", 1);
+						current_line_length++;
+					}
+					gb_file_write(&f, &part, 1);
+					current_line_length++;
+				}
+			}
```

Comment:
It might make sense to export it to a .json file, because it'll need more than just the .odin files in the future, maybe even the linking libraries and other things.

Reply:
I want to maintain the ability to output in Make format for compatibility with GCC and Clang, and existing build tools. But I agree that it is not a great format for new tools and will implement a way to output either as Make, or a JSON file.